src/main/scala/cromwell/engine/workflow/SingleWorkflowRunnerActor.scala

```diff
@@ -79,18 +85,19 @@ case class SingleWorkflowRunnerActor(source: WorkflowSourceFiles,
       log.info(s"$tag: workflow ID UUID($id)")
       workflowManager ! SubscribeToWorkflow(id)
 
-      if (getAbortJobsOnTerminate) {
-        Runtime.getRuntime.addShutdownHook(new Thread() {
-          override def run(): Unit = {
-            workflowManager ! WorkflowAbort(id)
-            log.info(s"$tag: Waiting for workflow $id to abort...")
-            while(stateName != Done)
-              Thread.sleep(1000)
-            log.info(s"$tag: Workflow $id aborted.")
-          }
-        })
+      if (abortJobsOnTerminate) {
+        sys.addShutdownHook {
+          workflowManager ! WorkflowAbort(id)
+          log.info(s"$tag: Waiting for workflow $id to abort...")
+          Await.ready(donePromise.future, Duration.Inf)
```

Comment:
As far as I understand the contract (and consistent with what I've observed), the JVM tries to execute all shutdown hooks and then exit.  So this actually needs to block, otherwise the JVM would shut down as soon as the code here finished without waiting for workflow(s) to abort.  I'll add a comment to this effect.